src/main/java/killjoy/task/Deadline.java

```diff
@@ -0,0 +1,32 @@
+package killjoy.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a task of type Deadline.
+ */
+public class Deadline extends Task {
+    private LocalDateTime by;
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd MMM yyyy HH:mm:ss");
```

Comment:
I feel "FORMATTER" is ambiguous. Looking at your code, its only purpose is to format the date in toString(). Consider:
```suggestion
    private static final DateTimeFormatter TO_STRING_FORMATTER = DateTimeFormatter.ofPattern("dd MMM yyyy HH:mm:ss");
```
Another small nit but consider moving FORMATTER to Task so you don't redefine it in every subclass of Task.